src/sentry/incidents/charts.py

```diff
@@ -0,0 +1,232 @@
+from datetime import datetime, timedelta
+from functools import reduce
+from typing import Any, List, Mapping, Optional, cast
+
+from django.utils import timezone
+
+from sentry.api import client
+from sentry.api.base import logger
+from sentry.api.serializers import serialize
+from sentry.api.serializers.models.alert_rule import AlertRuleSerializer
+from sentry.api.serializers.models.incident import DetailedIncidentSerializer
+from sentry.api.utils import get_datetime_from_stats_period
+from sentry.charts import generate_chart
+from sentry.charts.types import ChartType
+from sentry.incidents.logic import translate_aggregate_field
+from sentry.incidents.models import AlertRule, Incident, User
+from sentry.models import ApiKey, Organization
+from sentry.snuba.dataset import Dataset
+from sentry.snuba.entity_subscription import apply_dataset_query_conditions
+from sentry.snuba.models import SnubaQuery
+
+CRASH_FREE_SESSIONS = "percentage(sessions_crashed, sessions) AS _crash_rate_alert_aggregate"
+CRASH_FREE_USERS = "percentage(users_crashed, users) AS _crash_rate_alert_aggregate"
+SESSION_AGGREGATE_TO_FIELD = {
+    CRASH_FREE_SESSIONS: "sum(session)",
+    CRASH_FREE_USERS: "count_unique(user)",
+}
+
+
+API_INTERVAL_POINTS_LIMIT = 10000
+API_INTERVAL_POINTS_MIN = 150
+TIME_FORMAT = "%Y-%m-%dT%H:%M:%S"
+
+
+def incident_date_range(alert_rule: AlertRule, incident: Incident) -> Mapping[str, str]:
+    """
+    Retrieve the start/end for graphing an incident.
+    Will show at least 150 and no more than 10,000 data points.
+    This function should match what is in the frontend.
+    """
+    time_window_milliseconds = alert_rule.snuba_query.time_window * 1000
```

Comment:
Why do we convert these to ms? Later on this feeds into `timedelta(seconds=range / 2)`, won't that make the range too large?

Reply:
using ms because the frontend uses ms and its easier to do the same thing. the timedelta seconds should've been milliseconds.